lib/stm32wba/concurrent/STM32_WPAN/Target/linklayer_plat.c

```diff
@@ -0,0 +1,628 @@
+/* USER CODE BEGIN Header */
+/**
+  ******************************************************************************
+  * @file    linklayer_plat.c
+  * @author  MCD Application Team
+  * @brief   Source file for the linklayer plateform adaptation layer
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2022 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+
+#include "stm32wbaxx_hal.h"
+#include "stm32wbaxx_hal_conf.h"
+#include "stm32wbaxx_ll_rcc.h"
+
+#include "app_common.h"
+#include "app_conf.h"
+#include "linklayer_plat.h"
+#include "log_module.h"
+
+#ifndef __ZEPHYR__
+#if (CFG_LPM_LEVEL != 0)
+#include "stm32_lpm.h"
+#include "stm32_lpm_if.h"
+#endif /* (CFG_LPM_LEVEL != 0) */
+#endif
+
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+#ifndef __ZEPHYR__
+#define max(a,b) ((a) > (b) ? a : b)
+
+/* 2.4GHz RADIO ISR callbacks */
+void (*radio_callback)(void) = NULL;
+void (*low_isr_callback)(void) = NULL;
+
+/* RNG handle */
+extern RNG_HandleTypeDef hrng;
+
+/* Radio critical sections */
+static uint32_t primask_bit = 0;
+volatile int32_t prio_high_isr_counter = 0;
+volatile int32_t prio_low_isr_counter = 0;
+volatile int32_t prio_sys_isr_counter = 0;
+volatile int32_t irq_counter = 0;
+volatile uint32_t local_basepri_value = 0;
+
+/* Radio SW low ISR global variable */
+volatile uint8_t radio_sw_low_isr_is_running_high_prio = 0;
+#endif /* __ZEPHYR__ */
+
+/* Radio bus clock control variables */
+uint8_t AHB5_SwitchedOff = 0;
+uint32_t radio_sleep_timer_val = 0;
+
+/**
+  * @brief  Configure the necessary clock sources for the radio.
+  * @param  None
+  * @retval None
+  */
+void LINKLAYER_PLAT_ClockInit(void)
+{
+#ifndef __ZEPHYR__
```

Comment:
Use of `#ifndef __ZEPHYR__` inside `LINKLAYER_PLAT_` functions make the code difficult to review.
Prefer re-defining functions in zephyr repo as aloready done for part of them so we have all the zephyr adaptation in one place instead of having it splitted in multiples files as today.
This is harder to review, debug and maintain.